test/unit/prometheus-servicemonitor.bats

```diff
@@ -115,11 +115,53 @@ load _helpers
 @test "prometheus/ServiceMonitor-server: assertEndpoints TLS" {
   cd `chart_dir`
   local output=$( (helm template \
-      --show-only templates/prometheus-servicemonitor.yaml \
-      --set 'global.tlsDisable=false' \
-      --set 'serverTelemetry.serviceMonitor.enabled=true' \
-      . ) | tee /dev/stderr)
+    --show-only templates/prometheus-servicemonitor.yaml \
+    --set 'global.tlsDisable=false' \
+    --set 'serverTelemetry.serviceMonitor.enabled=true' \
+    .) | tee /dev/stderr)
 
   [ "$(echo "$output" | yq -r '.spec.endpoints | length')" = "1" ]
   [ "$(echo "$output" | yq -r '.spec.endpoints[0].port')" = "https" ]
 }
+
+@test "prometheus/ServiceMonitor-server: tlsConfig default" {
+  cd `chart_dir`
+  local output=$( (helm template \
+    --show-only templates/prometheus-servicemonitor.yaml \
+    --set 'serverTelemetry.serviceMonitor.enabled=true' \
+    .) | tee /dev/stderr)
+
+  [ "$(echo "$output" | yq -r '.spec.endpoints[0].tlsConfig.insecureSkipVerify')" = "true" ]
+}
+
+@test "prometheus/ServiceMonitor-server: tlsConfig override" {
+  cd `chart_dir`
+  local output=$( (helm template \
+    --show-only templates/prometheus-servicemonitor.yaml \
+    --set 'serverTelemetry.serviceMonitor.tlsConfig.ca=ca.crt' \
+    --set 'serverTelemetry.serviceMonitor.enabled=true' \
+    .) | tee /dev/stderr)
+
+  [ "$(echo "$output" | yq -r '.spec.endpoints[0].tlsConfig.ca')" = "ca.crt" ]
+}
+
+@test "prometheus/ServiceMonitor-server: bearerTokenFile default" {
+  cd `chart_dir`
+  local output=$( (helm template \
+    --show-only templates/prometheus-servicemonitor.yaml \
+    --set 'serverTelemetry.serviceMonitor.enabled=true' \
+    .) | tee /dev/stderr)
+
+  [ "$(echo "$output" | yq -r '.spec.endpoints[0] | has("bearerTokenFile")')" = "false" ]
+}
+
+@test "prometheus/ServiceMonitor-server: bearerTokenFile set" {
```

Comment:
@eyenx this test fails locally. It appears we have a misconfiguration in the relevant workflow file that stops it from being executed when a PR is opened from a fork.

Reply:
The test has been adjusted to use the `authorization` field